File: webservice/Frikollection_Api/Frikollection_Api/DTOs/Product/ProductDto.cs
using Frikollection_Api.DTOs.ProductExtension;
using Frikollection_Api.DTOs.ProductType;

namespace Frikollection_Api.DTOs.Product
{
    public class ProductDto
    {
        public string Name { get; set; } = null!;
        public string? Supertype { get; set; }
        public string? Subtype { get; set; }
        public DateOnly? Release { get; set; }
        public string? Status { get; set; }
        public int? ItemNumber { get; set; }
        public string? License { get; set; }
        public double? Width { get; set; }
        public double? Height { get; set; }
        public decimal? Value { get; set; }
        public string? SmallPicture { get; set; }
        public string? BigPicture { get; set; }
        public ProductTypeDto? ProductType { get; set; }
        public ProductExtensionDto? ProductExtension { get; set; }
    }
}

File: mobile_desktop_app/FrikollectionMobileDesktop/composeApp/src/commonMain/kotlin/org/example/frikollection_mobile_desktop/models/ProductDto.kt
package org.example.frikollection_mobile_desktop.models

import kotlinx.serialization.Serializable

@Serializable
data class ProductDto(
    val productId: String,
    val name: String,
    val supertype: String? = null,
    val subtype: String? = null,
    val release: String? = null, // Format "YYYY-MM-DD"
    val status: String? = null,
    val itemNumber: Int? = null,
    val license: String? = null,
    val width: Double? = null,
    val height: Double? = null,
    val value: Double? = null,
    val smallPicture: String? = null,
    val bigPicture: String? = null,
    val productType: ProductTypeDto? = null,
    val productExtension: ProductExtensionDto? = null,
    val tags: List<TagDto> = emptyList()
)